src/qibolab/instruments/qm.py

```diff
@@ -581,7 +581,11 @@ def fetch_results(result, ro_pulses):
                 shots = handles.get(f"{serial}_shots").fetch_all().astype(int)
             else:
                 shots = None
-            results[pulse.qubit] = results[serial] = ExecutionResults.from_components(ires, qres, shots)
+                results[pulse.qubit] = results[serial] = (
+                    AveragedResults.from_components(ires, qres)
+                    if average
+                    else ExecutionResults.from_components(ires, qres, shots)
+                )
```

Comment:
@andrea-pasquale if you use `platform.sweep(average=True)` then you should also get the shots (assuming a threshold and angle were provided). As it is now you are losing the shots.

This should be fixed if you move the `results[...]` assignment one indent back (outside the if).

Reply:
@stavros11 wait, I thought that if you average you don't get the shots (at least for qblox but I may be wrong).